privacy/anonymizing-your-mac-address.md

```diff
@@ -170,3 +172,48 @@ To disable MAC Randomizing if you find that a network connecting to does not lik
 - Also in `fedora-23` type `sudo systemctl disable macspoof@enp0s0`
 - Remove the service file `sudo rm /etc/systemd/system/macspoof@.service` in TemplateVM
 - Delete the package `sudo dnf remove macchanger`
+
+Method 2
+========
+
+An alternative approach is to use udev rules.
+
+This method does not require you to specify interface names, will apply to all interfaces, even those added after boot, and will retain the random MAC address through sleep/wake cycles.
+
+Install **macchanger** package in the TemplateVM backing sys-net.
+
+In that template create a udev rule using an editor of your choice.
+
+In /etc/udev/rules.d create the file 00-mac-spoof.rules with content:
+```
+SUBSYSTEM=="net", ACTION=="add", ATTR{type}=="1", RUN+="/usr/bin/macchanger -e  $name"
```

Comment:
> NM can access the original address and so the issue you identified doesnt arise. 

The question is *if* NM will use original address for `unmanaged-devices` setting. Make sure it will not mess with `vif*`. Is it still effective when you restart NM after that udev rule change the macchanger?

> (If you're wondering about this I see it's exactly the same as the Tails use of macchanger, so the NM issue isn't perceived to be a problem. See the refs linked to above.)

Tails doesn't have statically configured inter-VM interfaces...

> As to the setting for eth0 in VMs, I had looked at that, and hadn't realised that you could set the MAC - I have never been able to do so. When I try with qvm-prefs -s MAC blah I get an error "ERROR: wrong property name: 'MAC' "

`qvm-prefs -s ... mac`. Interesting why it is listed in uppercase.